numba/tests/npyufunc/test_ufuncbuilding.py

```diff
@@ -355,6 +355,75 @@ def test_guvectorize_invalid_layout(self):
         # (error message depends on Numpy version)
 
 
+class TestNEP13WithoutSignature(TestCase):
+
+    def test_all(self):
+
+        # note: no signatures specified
+        @vectorize(nopython=True)
+        def new_ufunc(hundreds, tens, ones):
+            return 100*hundreds + 10*tens + ones
+
+        # https://numpy.org/neps/nep-0013-ufunc-overrides.html
+        class NEP13Array:
+            def __init__(self, array):
+                self.array = array
+
+            def __array__(self):
+                return self.array
+
+            def tolist(self):
+                return self.array.tolist()
+
+            def __array_ufunc__(self, ufunc, method, *args, **kwargs):
+                if method != "__call__":
+                    return NotImplemented
+
+                return NEP13Array(ufunc(*[np.asarray(x) for x in args], **kwargs))
+
+        # give it integers
+        a = np.array([1, 2, 3], dtype=np.int64)
+        b = np.array([4, 5, 6], dtype=np.int64)
+        c = np.array([7, 8, 9], dtype=np.int64)
+
+        all_np = new_ufunc(a, b, c)
+        assert isinstance(all_np, np.ndarray)
```

Comment:
Our test code style is using `TestCase.assert*`. Can you change the `assert` to the corresponding `assert*` method?


(side-node: i wish we can use pytest but it still have problems due to pytest bytecode patching and we cannot reliably disable it)

Reply:
Of course! I used to use unittest and was fitting into this framework, but I had forgotten that it has its own assertion methods.